File: C#Solutions/problem12.cs
using System;



namespace AllProblems
{
    public static class Problem12
    {
        public static void Solve()
        {
            Console.WriteLine("Solution to Problem 12:Interger to Roman Convertion");
            
            Console.WriteLine("Enter the numbers to convert Roman Number");
            int num = int.Parse(Console.ReadLine());

            string roman = integerToRoman(num);
            
            Console.WriteLine("Roman Number: " + roman);
            
        }

        

        public static string integerToRoman(int num)
        {
           int n1;
        List<char> roman = new List<char>();

        while( num > 0){
    
            if (num >= 1000){
                n1 = num / 1000;
                num = num % 1000;
        
                for(int i = 0; i< n1; i++){
                    roman.Add('M');
                }
            }
            else if (num >= 100 && num < 1000 ){
                n1 = num / 100;
                num = num % 100;
                if(n1 < 4){
                    for(int i = 0; i< n1; i++){
                    roman.Add('C');
                     }
                }
                else if(n1 == 4){
                    roman.Add('C');
                    roman.Add('D');

                }
                else if(n1 == 5){
                    roman.Add('D');
        
                }
                else if (n1 > 5 && n1 < 9){
                    roman.Add('D');
                    for(int i = 6; i <= n1;i++){
                    roman.Add('C');

                    }
                 }
                else if(n1 == 9){
                    roman.Add('C');
                    roman.Add('M');
                }

            }
            else if (num >= 10 && num < 100){
                n1 = num / 10;
                num = num % 10;
                if(n1 < 4){
                    for(int i = 0; i< n1; i++){
                    roman.Add('X');
                    }
                }
                else if(n1 == 4){
                    roman.Add('X');
                    roman.Add('L');
                }
                else if(n1 == 5){
                    roman.Add('L');
        
                }
                else if (n1 > 5 && n1 < 9){
                    roman.Add('L');
                    for(int i = 6; i <= n1;i++){
                    roman.Add('X');

                    }
                }
                else if(n1 == 9){
                    roman.Add('X');
                    roman.Add('C');
                }

            }
            else if (num < 10){
                n1 = num;
                num = 0;
                if(n1 < 4){
                    for(int i = 0; i< n1; i++){
                    roman.Add('I');
                    }
                }
                else if(n1 == 4){
                    roman.Add('I');
                    roman.Add('V');
                }
                else if(n1 == 5){
                    roman.Add('V');
                }
                else if (n1 > 5 && n1 < 9){
                    roman.Add('V');
                    for(int i = 6; i <= n1;i++){
                    roman.Add('I');
                    }
                }
                else if(n1 == 9){
                    roman.Add('I');
                    roman.Add('X');
                }
            }
            else {
                break;
            }
        }
        string romanNumeral = new string(roman.ToArray());
        return romanNumeral;
            
        }

        
    }
}

File: python_solution/problem12.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Tue Sep  3 19:44:31 2024

@author: sharminalina
"""

class Problem12:
    def integerToRoman(self, num: int)-> str:
        roman =[]

        while num > 0:
            if num >= 1000:
                n2 = num // 1000
                num = num % 1000
                for i in range(n2):
                    roman.append('M')
            elif num >= 100:
                n2 = num // 100
                num = num % 100
                if n2 < 4:
                    for i in range(n2):
                        roman.append('C')
                elif n2 == 4:
                    roman.append('CD')
                elif n2 == 5:
                    roman.append('D')
                elif n2 > 5 and n2 < 9:
                    roman.append('D')
                    for i in range(5,n2):
                        roman.append('C')
                elif n2 == 9:
                    roman.append('CM')
            elif num >= 10:
                n2 = num // 10
                num = num % 10
                if n2 < 4:
                    for i in range(n2):
                        roman.append('X')
                elif n2 == 4:
                    roman.append('XL')
                elif n2 == 5:
                    roman.append('L')
                elif n2 > 5 and n2 < 9:
                    roman.append('L')
                    for i in range(5,n2):
                        roman.append('X')
                
                elif n2 == 9:
                    roman.append('XC')
            elif num > 0:
                n2 = num
                num = 0
            
                if n2 < 4:
                    for i in range(n2):
                        roman.append('I')
                elif n2 == 4:
                    roman.append('IV')
                elif n2 == 5:
                    roman.append('V')
                elif n2 > 5 and n2 < 9:
                    roman.append('V')
                    for i in range(5,n2):
                        roman.append('I')
                   
                
                elif n2 == 9:
                    roman.append('IX')
                else:
                    break
            
            else:
                break
            
                
        return ("".join(roman)) 

def main():
    
    # Read Integer array of height for calculating the area with most water
    
    print("Solution to Problem 12:Integer to Roman");
            
   
    
    num = int(input("Enter the number convert to Roman : "))
   

    # Find the maximum area of container with most water
    solution_prob12 = Problem12()
    romanNumber = solution_prob12.integerToRoman(num)

    # Print the result
    print(f"The Roman number of given integer : {romanNumber}")

if __name__ == "__main__":
    main()